lib/client.js

```diff
@@ -621,17 +644,31 @@ class Client extends EventEmitter {
     }
 
     if (this[kDestroyed]) {
-      process.nextTick(cb, new ClientDestroyedError(), null)
+      process.nextTick(callback, new ClientDestroyedError(), null)
       return false
     }
 
+    let called = false
+    function onDestroyed (err) {
+      if (!called) {
+        called = true
+        callback(err, null)
+      }
+    }
+
     this[kClosed] = true
+    this[kOnDestroyed].push(onDestroyed)
+
+    for (const request of this[kQueue].slice(this[kInflight])) {
+      if (request.signal) {
+        request.signal.on('error', onDestroyed)
```

Comment:
should this still wait for the socket to be destroyed before invoking the callback?